dtfs-central-api/src/v1/controllers/utilisation-report-service/upload-utilisation-report.controller.js

```diff
@@ -0,0 +1,45 @@
+const { saveUtilisationData } = require('../../../services/repositories/utilisation-data-repo');
+const { saveUtilisationReportDetails } = require('../../../services/repositories/utilisation-reports-repo');
+const {
+  validateUtilisationReportData,
+  validateMonth,
+  validateYear,
+  validateFilePath,
+} = require('../../validation/utilisation-report-service/utilisation-report-validator');
+
+const validatePayload = (reportData, month, year, filePath) => {
+  const validationErrors = validateUtilisationReportData(reportData);
+  const monthValidationError = validateMonth(month);
+  if (monthValidationError) validationErrors.push(monthValidationError);
+  const yearValidationError = validateYear(year);
+  if (yearValidationError) validationErrors.push(yearValidationError);
+  const filePathValidationError = validateFilePath(filePath);
+  if (filePathValidationError) validationErrors.push(filePathValidationError);
+  return validationErrors;
+};
+
+const postUtilisationReportData = async (req, res) => {
+  try {
+    const {
+      reportData, month, year, user, filePath
+    } = req.body;
+    const { bank } = user;
```

Comment:
Do we also need to validate that the `bank.id` isn't `*` (as it would be for user's with access to all banks, like Admins)? This has come up after discussion around whether Admins should be able to see the two GEF report nav items. It isn't hard-coded that admins have access to all banks though, so I guess this could technically be an issue for every user.

Re the nav bar issue, I've suggested that this be addressed in a separate ticket. Maybe that ticket could cover handling of access to multiple banks in other places too (like this). What do you think?

Reply:
Ah yes, ideally I don't want to have to deal with this as much. We should be only allowing admins to have * (which we don't atm) and I'm not sure we want to even support admins being able to 'upload a report' for a bank, aka they shouldn't really have access to the upload.

All in all I like your suggestion of yes let's tackle this separately and we'll need to catch up with DTFS team not just E2ETF team for this